### PR TITLE
dedupe 2.0.23 :snowflake:

### PR DESCRIPTION
**Pull request by lorepirri**
dedupe 2.0.23 :snowflake:

**Destination channel:** Snowflake

### Links

- [PKG-4748]
- dev_url (pypi): https://github.com/dedupeio/dedupe/tree/v2.0.23
- [v2.0.20...v2.0.23](https://github.com/dedupeio/dedupe/compare/v2.0.20...v2.0.23) diff : https://github.com/dedupeio/dedupe/compare/v2.0.20...v2.0.23
- conda-forge: https://github.com/conda-forge/dedupe-feedstock/blob/main/recipe/meta.yaml
- pypi: https://pypi.org/project/dedupe/2.0.23
- pypi inspector: https://inspector.pypi.io/project/dedupe/2.0.23

### Explanation of changes:

- reform recipe
- remove LICENCE from recipe
- patch out `dedupe-variable-datetime` which is actually an optional dependency

[PKG-4748]: https://anaconda.atlassian.net/browse/PKG-4748?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,39 +1,47 @@
...
+  sha256: d4f4e5bd777e3b8aeeac68eee5fbd36d4305cc8eac3115f778e3711a9dad8423
+  patches:
+    - patches/0001-fix-dependencies.patch
 
 build:
   number: 0
-  script: {{ PYTHON }} -m pip install . -vv
-  skip: true  # [win]
+  # skip s390x: no btrees
+  skip: true  # [s390x]
```
Comment (JamesRobertsonGames) [inline recipe/meta.yaml]:
```suggestion
  skip: true  # [s390x or py<37]
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,39 +1,47 @@
...
   build:
     - {{ compiler('c') }}
+    - patch     # [not win]
+    - m2-patch  # [win]
   host:
+    - python
     - cython
     - pip
-    - python
+    - setuptools >=63
```
Comment (JamesRobertsonGames) [inline recipe/meta.yaml:30]: Lock setuptools https://github.com/dedupeio/dedupe/blob/f72d4a161bfc66c9e1de9b39e2bd7e01bcad3c49/pyproject.toml#L54C1-L55C1
```suggestion
    - setuptools ==63
```

**Comment by skupr-anaconda** [comment]
As `Dedupe.io was shut down Jan 31, 2023.`, is it still relevant package to build/use?
See https://dedupe.io/

File: recipe/meta.yaml (hunk 2 above)
Comment (lorepirri) [reply recipe/meta.yaml:30]: @JamesRobertsonGames we prefer to keep the lower boundary instead of an exact pinning

**Comment by lorepirri** [comment]
> As `Dedupe.io was shut down Jan 31, 2023.`, is it still relevant package to build/use? See https://dedupe.io/

@skupr-anaconda it is a direct request of SF

**Review by skupr-anaconda — APPROVED** [review]

**Review by JamesRobertsonGames — APPROVED** [review]

**Pull request merged**